.github/workflows/release.yml

```diff
@@ -0,0 +1,60 @@
+name: Build TF Provider
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.3
+      - name: Install C toolchain
+        run: |
+          sudo apt-get update
+          sudo apt install -y gcc-x86-64-linux-gnu
+          mkdir bin
+      - name: Cache Binaries
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-binaries
+        with:
+          path: bin
+          key: ${{ github.sha }}
+          restore-keys: ${{ github.sha }}
+      - name: Build Linux binary with Boringcrypto
+        run: |
+          CC=x86_64-linux-gnu-gcc CGO_ENABLED=1 GOARCH=amd64 GOOS=linux GOEXPERIMENT=boringcrypto \
+          go build -o bin/terraform-provider-azurerm.linux.amd64 .
+      - name: Verify Boringcrypto
+        run: |
+          go run rsc.io/goversion@master -crypto bin/terraform-provider-azurerm.linux.amd64 | grep -q '(boring crypto)'
```

Comment:
Are we 100% sure this certifies the crypto as boring? We've been burned here before... if we're sure, no worries, otherwise let's bust out our trusty `go tool nm`

Reply:
I've had a look at the implementation and it's a very similar check to the `go tool nm` we had before: https://github.com/rsc/goversion/blob/3a30cee7003ec5dc0903682e79b3059c8e101952/version/read.go#L60

But that implementation looks more solid & thorough than the `go tool` & grep chain 😄